lib/puppet/defaults.rb

```diff
@@ -1010,6 +1010,15 @@ def self.default_diffargs
           end
         end
     },
+    :ca_url_prefix => {
+        :default => "/puppet-ca",
+        :desc    => "The prefix at which the puppet certificate authority API is mounted.",
+        :hook    => proc do |value|
+          if !value.start_with?("/")
+            Puppet[:ca_url_prefix] = "/#{value}"
```

Comment:
A conversation with @nwolfe about the previous PR brought up the potential issue with this and the `master_url_prefix` that if a user specifies a prefix with a slash in it, they may end up with really confusing errors because `IndirectedRoutes` does a split on `/` (https://github.com/puppetlabs/puppet/pull/3464/files#diff-20dc5688c93b62d47f42bf0a835001ecR70). Would it make sense to raise an error for `ca_url_prefix` and `master_url_prefix` if a prefix contains a slash or other illegal characters?